libtenzir/src/data_builder.cpp

```diff
@@ -0,0 +1,1404 @@
+}
+
+auto basic_parser(std::string_view s, const tenzir::type* seed)
+  -> detail::data_builder::data_parsing_result {
+  if (seed) {
+    return basic_seeded_parser(s, *seed);
+  }
+  if (s.empty()) {
+    return {};
+  }
+  auto res = best_effort_parser(s);
+  if (res) {
+    return std::move(*res);
+  } else {
+    return {};
+  }
+}
+
+auto non_number_parser(std::string_view s, const tenzir::type* seed)
+  -> detail::data_builder::data_parsing_result {
+  if (seed) {
+    return data_builder::basic_seeded_parser(s, *seed);
+  }
+  if (s.empty()) {
+    return tenzir::data{std::string{}};
+  }
+  auto res = tenzir::data{};
+  if (parse_as_data<bool_type, time_type, duration_type, subnet_type, ip_type>(
+        s, res)) {
+    return res;
+  }
+  return {};
+}
+
+namespace {
+
+template <typename Field, typename Tenzir_Type>
+struct index_regression_tester {
+  constexpr static auto index_in_field
+    = caf::detail::tl_index_of<field_type_list, Field>::value;
+  constexpr static auto tenzir_type_index = Tenzir_Type::type_index;
+
+  static_assert(index_in_field == tenzir_type_index);
+
+  using type = void;
+};
+
+static_assert(caf::detail::tl_size<field_type_list>::value
+                == caf::detail::tl_size<data::types>::value + 1,
+              "The could should match up, apart from the lack of `enriched` in "
+              "`tenzir::data`");
+
+[[maybe_unused]] consteval void test() {
+  (void)index_regression_tester<caf::none_t, null_type>{};
+  (void)index_regression_tester<int64_t, int64_type>{};
+  (void)index_regression_tester<uint64_t, uint64_type>{};
+  (void)index_regression_tester<double, double_type>{};
+  (void)index_regression_tester<duration, duration_type>{};
+  (void)index_regression_tester<time, time_type>{};
+  (void)index_regression_tester<std::string, string_type>{};
+  (void)index_regression_tester<ip, ip_type>{};
+  (void)index_regression_tester<subnet, subnet_type>{};
+  (void)index_regression_tester<enumeration, enumeration_type>{};
+  (void)index_regression_tester<node_list, list_type>{};
+  (void)index_regression_tester<map_dummy, map_type>{};
+  (void)index_regression_tester<node_record, record_type>{};
+  (void)index_regression_tester<blob, blob_type>{};
+}
+} // namespace
+
+auto node_base::is_dead() const -> bool {
+  return state_ == state::dead;
+}
+auto node_base::is_alive() const -> bool {
+  return state_ == state::alive;
+}
+auto node_base::affects_signature() const -> bool {
+  switch (state_) {
+    case state::alive:
+    case state::sentinel:
+      return true;
+    case state::dead:
+      return false;
+  }
+  TENZIR_UNREACHABLE();
+}
+
+auto node_record::try_field(std::string_view name) -> node_object* {
+  auto [it, inserted] = lookup_.try_emplace(name, data_.size());
+  if (not inserted) {
+    return &data_[it->second].value;
+  }
+  TENZIR_ASSERT(data_.size() <= 20'000, "Upper limit on record size reached.");
+  return &data_.emplace_back(it->first).value;
+}
```

Comment:
This is reachable for my suricata data set, but only when using `--schema-only`. `--merge` and nothing works as expected.